src/Symfony/Component/CssSelector/Tests/CssSelectorConverterTest.php

```diff
@@ -33,4 +33,46 @@ public function testCssToXPathXml()
 
         $this->assertEquals('descendant-or-self::H1', $converter->toXPath('H1'));
     }
+
+    public function testParseExceptions()
+    {
+        $converter = new CssSelectorConverter();
+
+        try {
+            $converter->toXPath('h1:');
+            $this->fail('->parse() throws an Exception if the css selector is not valid');
+        } catch (\Exception $e) {
+            $this->assertInstanceOf('\Symfony\Component\CssSelector\Exception\ParseException', $e, '->parse() throws an Exception if the css selector is not valid');
+            $this->assertEquals('Expected identifier, but <eof at 3> found.', $e->getMessage(), '->parse() throws an Exception if the css selector is not valid');
```

Comment:
This should be turned into a ``@expectedException`` IMO (and a ``@expectedExceptionMessage`` instead of catching all exceptions (including from ``->fail()``) and then asserting